runtime/parser/keyword.go

```diff
@@ -75,6 +75,22 @@ const (
 	KeywordPub         = "pub"
 	KeywordPriv        = "priv"
 	KeywordInclude     = "include"
+	KeywordTry         = "try"
+	KeywordCatch       = "catch"
+	KeywordFinally     = "finally"
+	KeywordGoto        = "goto"
+	KeywordConst       = "const"
+	KeywordExport      = "export"
+	KeywordThrow       = "throw"
+	KeywordThrows      = "throws"
+	KeywordRequires    = "requires"
+	KeywordWhere       = "where"
+	KeywordFinal       = "final"
+	KeywordInternal    = "internal"
+	KeywordTypealias   = "typealias"
```

Comment:
This one seems a bit odd to me, why reserve `typealias` but not `type`?

Reply:
We can also reserve `type`. I just tried to come up with a list, looking at reserved keywords of Swift, JS, Java, etc. and hadn't thought of `type`. It's not intentionally omitted

Reply:
I think `type` is commonly used to create a type alias (e.g. Go, TypeScript), so we should reserve it just to be safe

Reply:
I tried adding `type`, but it is used in the standard library, e.g. in `forEachStored` as an argument label. I've added it to the soft keywords set.